ycmd/tests/testdata/general_fallback/lang_c.c

```diff
@@ -0,0 +1,85 @@
+// Identifier completion makes sense for duck-typing.
+//
+// The purpose of this test is to both demonstrate the use-case and test the
+// functionality. Completions of "A.a_parameter" using the identifier
+// completer feels natural, whereas offering no suggestions feels broken
+
+typedef struct {
+  int a_parameter;
+  int another_parameter;
+} TTest;
+
+typedef struct {
+  int another_int;
+  int and_a_final_int;
+} TTest_Which_Is_Not_TTest;
+
+static void do_something( int );
+static void do_another_thing( int );
+
+// TESTCASE1: use of . on macro parameter. Macros used this way are compile-time
+// duck-typing. Note this in this particular instance, libclang actually
+// offers semantic completions if there is only call to this macro
+// (e.g. DO_SOMETHING_TO( a_test ) - clever little shrew), so we don't call it
+
+/*       1         2         3         4
+1234567890123456789012345678901234567890123456789 */
```

Comment:
I'm not crazy enough to feel strongly about them. Feel free to leave them in. :)